pandas/core/construction.py

```diff
@@ -475,13 +475,8 @@ def sanitize_array(data, index, dtype=None, copy=False, raise_cast_failure=False
 
         if is_object_dtype(subarr.dtype) and not is_object_dtype(dtype):
             inferred = lib.infer_dtype(subarr, skipna=False)
-            if inferred == "period":
-                from pandas.core.arrays import period_array
-
-                try:
-                    subarr = period_array(subarr)
-                except IncompatibleFrequency:
-                    pass
+            if inferred in {"interval", "period"}:
```

Comment:
Would this path fail in general for extension scalars? e.g. Do Categoricals also fail here?

Reply:
`Categorical` shouldn't hit this code path since they need to be specified with a dtype, either by passing in a `Categorial` itself as the data, or a `CategoricalDtype` for the dtype, which should prevent the `if` block containing this code from being entered.

Looks like lists of tz-aware `Timestamp` objects are fine as well, as there is a `maybe_cast_to_datetime` earlier on that catches them.  This does suggest that maybe this logic could be condensed, so I'll look into that next evening or so.